src/persist-client/src/internal/state_versions.rs

```diff
@@ -271,16 +274,13 @@ impl StateVersions {
     /// copy of all or recent live diffs, to avoid fetches where possible.
     ///
     /// Panics if called on an uninitialized shard.
-    pub async fn fetch_current_state<K, V, T, D>(
+    pub async fn fetch_current_state<T>(
         &self,
         shard_id: &ShardId,
         mut live_diffs: Vec<VersionedData>,
-    ) -> Result<State<K, V, T, D>, Box<CodecMismatch>>
+    ) -> Result<UncheckedState<T>, Box<CodecMismatch>>
```

Comment:
this one is _so_ clunky, it's the one thing I really dislike about this approach (of course, there's always one). the `Box<CodecMismatch>` is for a T mismatch and the UncheckedState is for K, V, D